services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/service/MongoDbBasedTenantService.java

```diff
@@ -419,32 +375,9 @@ public void stop(final Promise<Void> stopPromise) {
         return findTenantPromise.future()
                 .compose(tenantJsonResult -> Optional.ofNullable(tenantJsonResult)
                         .map(tenantJson -> Future.succeededFuture(tenantJson.mapTo(TenantDto.class)))
-                        .orElse(Future.failedFuture(new NullPointerException()))
-                        .recover(error -> {
-                            if (findQuery.containsKey(RegistrationConstants.FIELD_PAYLOAD_TENANT_ID)) {
-                                return Future.failedFuture(
-                                        new ClientErrorException(HttpURLConnection.HTTP_NOT_FOUND,
-                                                String.format("Tenant [%s] not found.",
-                                                        findQuery.getString(
-                                                                RegistrationConstants.FIELD_PAYLOAD_TENANT_ID))));
-                            }
-                            return Future.failedFuture(
-                                    new ClientErrorException(HttpURLConnection.HTTP_NOT_FOUND, "Tenant not found"));
-                        }));
+                        .orElse(Future.failedFuture(new ClientErrorException(HttpURLConnection.HTTP_NOT_FOUND))));
```

Comment:
Since it is a failed future, how about also adding` String.format("Tenant [%s] not found.", tenantId)` as a next argument to the `ClientErrorException`?

Reply:
done